pandas/core/algorithms.py

```diff
@@ -944,14 +944,64 @@ def select_n_frame(frame, columns, n, method, keep):
     -------
     nordered : DataFrame
     """
-    from pandas.core.series import Series
+    from pandas import Int64Index
     if not is_list_like(columns):
         columns = [columns]
     columns = list(columns)
-    ser = getattr(frame[columns[0]], method)(n, keep=keep)
-    if isinstance(ser, Series):
-        ser = ser.to_frame()
-    return ser.merge(frame, on=columns[0], left_index=True)[frame.columns]
+    for column in columns:
+        dtype = frame[column].dtype
+        if not issubclass(dtype.type, (np.integer, np.floating, np.datetime64,
```

Comment:
though, I think you call EACH column (as a Series) right? so this check is *already* there (well not 100% that is actually checks things).

Or you *may* skip a column if no dupes? 

What I am trying to say is that I want to defer the dtype checks to the other algos if possible (and not do it here).